### ISSUE TITLE
Issue with extension activation

### ISSUE THREAD

**Issue by bobbrow** [microsoft/vscode-cmake-tools#1794]
Can this in any way be disabled over a configuration? 
We are having currently the issue that the cmakelist.txt will be created after the extension is loaded. But then the extension bar is never loaded.

_Originally posted by @JanHildenbrandBosch in https://github.com/microsoft/vscode-cmake-tools/issues/1578#issuecomment-824005940_

**Comment by bobbrow** [comment]
Can you provide more details as to what you are expecting?  I don't understand your description 100%.  Is this the case where you open an empty folder, add a CMakeLists.txt to the folder, but the CMake extension doesn't activate?

If that's what you're reporting, I'm not sure if there's an issue for this yet so we can use this to track it.  I have seen this behavior before.  As a workaround you can activate the extension by running the "CMake: Configure" command.

**Comment by JanHildenbrandBosch** [comment]
@bobbrow  Sorry for not creating a new issue, I thought it was directly related to the mentioned issue. I think you got my point. Let me elaborate a bit:

In our dev environment (which is inside a docker container) we set the cmake source directory (```"cmake.sourceDirectory" : "/src"```) to a non-existing path for example "/src". Because of some performance reasons we copy our source code from the mounted directory to "/src" before each cmake call. After recreating the environment (which can happen somewhat frequently) the "/src" folder is empty, and therefore the cmake bar at the bottom is vanished since version 1.7.0, in previous versions this was not a problem.

I'd would be nice for us to configure, that the cmake extension is not automatically disabling if there is no CMakeLists.txt present.
We got already a workaround by initially setting up the "/src" folder before starting the extension, but I was curious, whether this can also be just configured in the extension.

**Comment by andreeis** [comment]
@JanHildenbrandBosch, even if the bottom status bar is hidden (because we detect that we don't have a valid CMake project) the command "CMake: Configure" is still available in the palette. What did you do before when the status bar remained visible? You were pressing one of the buttons anyway, to trigger a configure, right? Now you would have to run the configure command from the palette. Is this too inconvenient?
Of course we can add a setting but considering that the majority of our users prefer having the UI elements hidden for non CMake projects and that we already have numerous settings, we just think twice before adding a new one. Let us know how such new setting would help your development scenario.
We will also make sure the full activation happens automatically when a CMakeLists.txt appears (which is why we keep this GitHub report open).

**Comment by JanHildenbrandBosch** [comment]
It's fine from my side, I understand the concern of hiding the UI and adding too much configuration.

For me personally it is fine to use the command palette, but we maintain one devcontainer for a lot of different developers with different technical background. The status bar is very easy to use and the change in 1.7.0 disrupted their known workflow for some developers. 

Our "workaround" works fine, so we can live without the setting :).

**Comment by andreeis** [comment]
Ok, thank you for the follow up. Then we will keep this report for making sure we can transition easily from a non CMake project to a suddenly appearing CMakeLists.txt.

**Comment by recursivenomad** [comment]
This issue is still relevant - I have been manually creating CMakeLists.txt in my projects, and I find myself looking for the CMake Tools interface to begin building, not thinking that I must summon the command palette to initiate this.  I would appreciate the extension to automatically activate as suggested above, or at least have the option of manually triggering the extension to configure (and activate) via an always-on GUI element.

**Comment by recursivenomad** [comment]
Happy to report that sometime in the past year, this issue has been resolved 🙂 Creating `CMakeLists.txt` in a new workspace/directory initializes the CMake Tools extension automatically, confirmed with CMake Tools version `1.20.53` on my machine 🎉

**Comment by Amy-Li03** [comment]
Hi @recursivenomad, we are very happy that your issue has been resolved, thanks for help us build a better CMake Tools!
